### DISCUSSION TITLE
(How to?) add a direct call to `citar-file-open-external` from citar's "menu"

### DISCUSSION THREAD

**Discussion by JonatanSahar** [emacs-citar/citar#828] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I often want to open a citation's pdf in an external program. currently I've added `citar-file-open-external` to embark actions on the file as it appears after I call `citar-open`, but this is quite a lot of keystrokes. 

**Describe the solution you'd like**
A function that calls `citar-file-open-external` on the citation at point, or a snippet that shows how to do so

**Describe alternatives you've considered**
I've bound  `citar-file-open-external`  in `citar-citation-map`, but I don't know how to pass it the citekey it needs in order to work.

**Comment by bdarcus** [comment, marked as answer]
@JonatanSahar 

> I often want to open a citation's pdf in an external program. currently I've added citar-file-open-external to embark actions on the file as it appears after I call citar-open, but this is quite a lot of keystrokes.
>
> Describe the solution you'd like
> A function that calls citar-file-open-external on the citation at point, or a snippet that shows how to do so

Just want to clarify something first.

If what you want is the latter, why is the former "quite a lot of keystrokes"?

And are you aware of this defcustom? Could you not just set PDFs to use `citar-file-open-external` by default?

https://github.com/emacs-citar/citar/blob/885b86f6733fd70f42c32dd7791d3447f93db990/citar-file.el#L39

**Reply by JonatanSahar** [reply]
that's a great solution, thank you!
what I meant about keystrokes is that I'd have liked to hit RET, then a letter (e.g., `f`) to open the PDF externally, and in the solution that I had, I had to go RET, then o, go down to the file, then hit `c-;` for embark and finally `f`.
I'd still like to be able to open the pdf directly, but this is totally good enough for me.
